src/extensions/extension-loader/extension-loader.ts

```diff
@@ -312,35 +317,58 @@ export class ExtensionLoader {
 
             if (!LensExtensionClass) {
               this.nonInstancesByName.add(extension.manifest.name);
-              continue;
+
+              return null;
             }
 
             const instance = this.dependencies.createExtensionInstance(
               LensExtensionClass,
               extension,
             );
 
-            const loaded = instance.enable(register).catch((err) => {
-              logger.error(`${logModule}: failed to enable`, { ext: extension, err });
-            });
-
-            loadingExtensions.push({
+            return {
+              extId,
+              instance,
               isBundled: extension.isBundled,
-              loaded,
-            });
-            this.instances.set(extId, instance);
+              activated: instance.activate(),
```

Comment:
Shouldn't be awaited?
```suggestion
              activated: await instance.activate(),
```

Reply:
That's not needed, as I want to store a promise to the object, then await them all, and go to the next step.

If I add `await` here, I need to add `async` to the `map` callback, which makes it an array of promises.